README.md

```diff
@@ -3,59 +3,332 @@
 [![crates.io](https://img.shields.io/crates/v/cargo-dist.svg)](https://crates.io/crates/cargo-dist) [![docs](https://docs.rs/cargo-dist/badge.svg)](https://docs.rs/cargo-dist)
 ![Rust CI](https://github.com/axodotdev/cargo-dist/workflows/Rust%20CI/badge.svg?branch=main)
 
-`cargo build` but For Building Final Distributable Artifacts.
+`cargo build` but For Building Final Distributable Artifacts and uploading them to an archive.
+
+The Big Idea of cargo-dist is that we want to streamline all the steps of providing prebuilt binaries
+for a rust project. This includes:
+
+* Generating your "Cut A Release" Github CI for you
+* Picking good build flags for a "production" release
+* Uploading all the resulting artifacts to a Github Release™️
+* Generating a machine-readable manifest so other tools can understand the Github Release™️
+
+Even though cargo-dist is primarily a tool for building and packaging applications, we put a fair amount of effort into
+Generating Your CI Scripts For You because we want to be able to run things locally and know what the CI *should* do without actually running it. It also helps avoid needless vendor lock-in -- in an ideal world, migrating from Github to Gitlab or your own personal infra would be just one invocation of cargo-dist away!
+
+That said, the current version is Very Very Unstable And Experimental and really only works with Github CI and Github Releases™️!
+
+* [Way-Too-Quick Start](#way-too-quick-start)
+* [Installation](#installation)
+  * [Build From Source With Cargo](#build-from-source-with-cargo)
+  * [Install Prebuilt Binaries With Cargo-Binstall](#install-prebuilt-binaries-with-cargo-binstall)
+  * [Download From Github Releases](#download-from-github-releases)
+* [Setup](#setup)
+  * [Configuring Installers](#configuring-installers)
+  * [Configuring Targets](#configuring-targets)
+* [Usage (CI)](#usage-ci)
+* [Usage (Local Builds)](#usage-local-builds)
+* [Concepts](#concepts)
+* [Build Flags](#build-flags)
+* [Compatibility With Other Tools](#compatibility-with-other-tools)
+* [Contributing](#contributing)
+  * [Updating Snapshots](#updating-snapshots)
+  * [Cutting Releases](#cutting-releases)
+
+
+
+# Way-Too-Quick Start
+
+```sh
+# install tools
+cargo install cargo-dist
+cargo install cargo-release
+
+# one-time setup
+cargo dist init --ci=github
+
+# cut a release
+cargo release 0.1.0
+```
+
+That's gonna do a whole bunch of stuff you might not have expected (but I left off the --execute flag from `cargo-release` so you won't actually break anything if you really did just copy paste that 😇). Read the rest of the docs to know what it does!
+
+
+
+
+
+# Installation
+
+## Build From Source With Cargo
```

Comment:
do we want to also show the installers you created?

Reply:
I don't want to advise those right now, because they put cargo in a pseudo-corrupt state where `cargo dist` runs but `cargo uninstall cargo-dist` says "there's no such thing" which is uhhh bad for anything but CI.

Reply:
Changed my mind and added them, and actually demoted the two options that Literally Won't Work Because It's Not Published On Crates.io.